mount/mount_setattr.go

```diff
@@ -0,0 +1,89 @@
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package mount
+
+import (
+	"runtime"
+	"syscall"
+	"unsafe"
+)
+
+// There is no support of mount_setattr in golang
+// "golang.org/x/sys/unix" package and even in glibc
+
+// Attr - configuration structure for SetAttr()
+type Attr struct {
+	attrSet     uint64
+	attrClr     uint64
+	propagation uint64
+	userNs      uint64
+}
+
+// AT_RECURSIVE - Apply changes to entire subtree
+const AT_RECURSIVE = 0x8000 //nolint
+
+const (
+	// MOUNT_ATTR_RDONLY - Mount read-only
+	MOUNT_ATTR_RDONLY = 0x00000001 //nolint
+	// MOUNT_ATTR_NOSUID - Ignore suid and sgid bits
+	MOUNT_ATTR_NOSUID = 0x00000002 //nolint
+	// MOUNT_ATTR_NODEV - Disallow access to device special files
+	MOUNT_ATTR_NODEV = 0x00000004 //nolint
+	// MOUNT_ATTR_NOEXEC - Disallow program execution
+	MOUNT_ATTR_NOEXEC = 0x00000008 //nolint
+	// MOUNT_ATTR__ATIME - Setting on how atime should be updated
+	MOUNT_ATTR__ATIME = 0x00000070 //nolint
+	// MOUNT_ATTR_RELATIME - Update atime relative to mtime/ctime
+	MOUNT_ATTR_RELATIME = 0x00000000 //nolint
+	// MOUNT_ATTR_NOATIME - Do not update access times
+	MOUNT_ATTR_NOATIME = 0x00000010 //nolint
+	// MOUNT_ATTR_STRICTATIME - Always perform atime updates
+	MOUNT_ATTR_STRICTATIME = 0x00000020 //nolint
+	// MOUNT_ATTR_NODIRATIME - Do not update directory access times
+	MOUNT_ATTR_NODIRATIME = 0x00000080 //nolint
+	// MOUNT_ATTR_IDMAP - Idmap mount to @userns_fd in struct mount_attr
+	MOUNT_ATTR_IDMAP = 0x00100000 //nolint
+)
+
+func setAttrSyscallNr() int {
+	switch runtime.GOARCH {
+	case "mips", "mipsle":
+		return 4441
+	case "mips64", "mips64le":
+		return 5441
+	default:
+		return 441
+	}
+}
```

Comment:
The syscall number changes in the upstream version. They are X442 -> https://github.com/brauner/mount-idmapped/blob/master/mount-idmapped.c#L70. Are you also covering all the architectures as it's done in the example above?

Reply:
Updated mount_setattr() system call number. Rechecked/fixed arch support.